test/driver.cpp

```diff
@@ -145,9 +164,19 @@ void processCmdLine(std::vector<std::string_view> arguments)
    }
    catch (const std::string & msg)
    {
-      std::cout << msg << '\n'; //TODO: print in appropriate stream based in cmd-line
+      (outputFilename.empty() ? std::cout : fileOut )<< msg << '\n';
    }
 
+   //if print summary option is enabled
    if (printSummary)
       summarizeTests();
+}
+
+//Replace all instances of a substring with a different substring
+void replace_all(string& str, const string& substr, 
+   const string& new_substr)
+{
+   string::size_type pos = str.find(substr);
+   for (; pos != string::npos; pos = str.find(substr, pos + substr.size()))
+      str.replace(pos, 4, new_substr);
```

Comment:
I believe this function definition will change based on my earlier observations about generalizing it, but it still needs to be correct in its current state: the last line of the function cannot use the fixed size `4`.

```suggestion
void replace_all(std::string& str, std::const string& substr, 
   const std::string& new_substr)
{
  auto pos = str.find(substr);
  auto substr_size = substr.size();
   for (; pos != std::string::npos; pos = str.find(substr, pos + substr_size))
      str.replace(pos, substr_size, new_substr);
```